common/env_s3.h

```diff
@@ -0,0 +1,198 @@
+/// Copyright 2018 Pinterest Inc.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+/// http://www.apache.org/licenses/LICENSE-2.0
+
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+
+//
+// @author ghan (ghan@pinterest.com)
+//
+
+#pragma once
+
+#include <algorithm>
+#include <stdio.h>
+#include <time.h>
+#include <iostream>
+#include <mutex>
+#include <pthread.h>
+
+#include "common/s3util.h"
+#include "common/sys_time.h"
+#include "rocksdb/env.h"
+#include "rocksdb/status.h"
+
+namespace common {
```

Comment:
does it make more sense to put it in rocksdb namespace? It may help reduce some code in the implementation code later.

Reply:
good point